### DISCUSSION TITLE
Does the Hybrid work with a Load Balanced Server Farm?

### DISCUSSION THREAD

**Discussion by timschwallie** [dotnetcore/EasyCaching#449] [General]
I'm not seeing anything in the documentation regarding implementations in a Server Farm.
The Servers/Applications would be load balanced using, for example a distributed database like Redis.

When using Hybrid, would a change migrate to the other Clients using the distributed database?

**Comment by catcherwong** [comment]
> I'm not seeing anything in the documentation regarding implementations in a Server Farm.
The Servers/Applications would be load balanced using, for example a distributed database like Redis.

EasyCaching is currently positioned as the abstraction layer of cache operations. It may not be able to implement distributed cache servers like Redis in the short term.

> When using Hybrid, would a change migrate to the other Clients using the distributed database?

I'm not sure exactly what this means.

**Reply by timschwallie** [reply]
Think I've got this.
In a Web Farm with a Load Balancer, I'm pretty sure I have an answer.

The Redis server needs to know the individual clients. A Load Balancer can confuse this.

So this is actually on Redis Client to ensure the connections are made and the Load Balance to ensure the connections are maintained per Hybrid participating client, toss in a little Networking.

If Redis is on same side as Web Farm (behind the Load Balancer), should be good to go for Hybrid.

If Redis is on outside side of Load Balancer and Web Farm, might need to check network configurations and Load Balancer configurations. This check is to ensure the Redis Pub/Sub talks to each Client participating in the Hybrid, as opposed to the Load Balancer sending all pub/sub communication to the same Client.
